src/lib/nameOldEigenQueries.ts

```diff
@@ -3,9 +3,9 @@ import { RequestHandler } from "express"
 
 export const nameOldEigenQueries: RequestHandler = (req, _res, next) => {
   const agent = req.headers["user-agent"]
-  if (agent && agent.includes("Eigen") && req.body.query) {
+  if (agent && agent.includes("Eigen")) {
     const { query } = req.body as { query: string }
-    if (!query.startsWith("query ")) {
+    if (query && !query.startsWith("query ")) {
```

Comment:
I think this check should be the very first one in this function. There’s no need to do string contents checking on all incoming requests when we can already cut down many of them from the get go. So:

```ts
if (req.body.query && agent && agent.includes("Eigen") && req.body.query) {
  ...
}
```

Reply:
It’s the little perf things that make up the whole